dbt/adapters/duckdb/environments.py

```diff
@@ -41,50 +45,101 @@ def handle(self):
     def cursor(self):
         raise NotImplementedError
 
-    def close(self):
+    def submit_python_job(self, handle, parsed_model: dict, compiled_code: str) -> AdapterResponse:
         raise NotImplementedError
 
+    def close(self, cursor):
+        raise NotImplementedError
 
-class LocalEnvironment(Environment):
-    def __init__(self, credentials: DuckDBCredentials):
-        self.creds = credentials
-        self.conn = duckdb.connect(credentials.path, read_only=False)
+    @classmethod
+    def initialize_db(cls, creds: DuckDBCredentials):
+        config = creds.config_options or {}
+        conn = duckdb.connect(creds.path, read_only=False, config=config)
 
         # install any extensions on the connection
-        if credentials.extensions is not None:
-            for extension in credentials.extensions:
-                self.conn.execute(f"INSTALL '{extension}'")
+        if creds.extensions is not None:
+            for extension in creds.extensions:
+                conn.execute(f"INSTALL '{extension}'")
 
         # Attach any fsspec filesystems on the database
-        if credentials.filesystems:
+        if creds.filesystems:
             import fsspec
 
-            for spec in credentials.filesystems:
+            for spec in creds.filesystems:
                 curr = spec.copy()
                 fsimpl = curr.pop("fs")
                 fs = fsspec.filesystem(fsimpl, **curr)
-                self.conn.register_filesystem(fs)
+                conn.register_filesystem(fs)
 
         # attach any databases that we will be using
-        if credentials.attach:
-            for entry in credentials.attach:
-                attachment = Attachment(**entry)
-                self.conn.execute(attachment.to_sql())
+        if creds.attach:
+            for attachment in creds.attach:
+                conn.execute(attachment.to_sql())
+        return conn
 
-    def handle(self):
-        return DuckDBConnectionWrapper(self)
-
-    def cursor(self):
+    @classmethod
+    def initialize_cursor(cls, creds, cursor):
         # Extensions/settings need to be configured per cursor
-        cursor = self.conn.cursor()
-        for ext in self.creds.extensions or []:
+        for ext in creds.extensions or []:
             cursor.execute(f"LOAD '{ext}'")
-        for key, value in self.creds.load_settings().items():
+        for key, value in creds.load_settings().items():
             # Okay to set these as strings because DuckDB will cast them
             # to the correct type
             cursor.execute(f"SET {key} = '{value}'")
```

Comment:
Might be worth escaping quotes here?

Reply:
I don't understand?

Reply:
(Like, which quotes should I be escaping?)

Reply:
In the case that someone passes a string containing single quotes I mean - they should be doubled up to escape them. It's not directly relevant to this pr, just figured it was worth mentioning